### PR TITLE
[example][fundamental] Use extra text file to declare tutorial workflow monitor resources

### PR DESCRIPTION
**Pull request by zhengfeiwang**
# Description

This PR targets to make workflow(s) for example(s) can be triggered correctly, so that contributor does not need to wait for unnecessary workflows not related to him/her.

The expected listen paths should have:

- working directory (where the notebook or markdown file locates)
- declared resource(s)
- corresponding workflow YAML file
- `examples/requirements.txt` - trick that makes release verification process convenient
- `examples/connections/azure_openai.yml` - fall back as it's the most basic and common connection

The major change list contains:

- `scripts/readme`: add logic to resolve resources from metadata
- `examples/tutorials`: declare resources in metadata
- `.github/workflows`: update workflows following new generation logic
- `examples/CONTRIBUTING.md`: update example contributing doc

Some side changes:

- Fix wrong workflow listen path for additional include and symbol link flows, as they have dependency to flow web classification
- Fix missing PEP 8 link in example contributing doc

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: examples/tutorials/flow-in-pipeline/pipeline.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+examples/flows/standard/web-classification
```
Comment (crazygao) [inline examples/tutorials/flow-in-pipeline/pipeline.txt:1]: this also depend on web-classification
and web classification depend on azure_openai.yml
however a chane in azure_openai.yml doesn't affect this pipeline.

Comment (crazygao) [reply examples/tutorials/flow-in-pipeline/pipeline.txt:1]: same with requirements.txt

Comment (zhengfeiwang) [reply examples/tutorials/flow-in-pipeline/pipeline.txt:1]: good catch, and make sense; have updated to always inject `examples/connections/azure_openai.yml`

File: examples/tutorials/flow-deploy/create-service-with-flow/resources.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+examples/tutorials/flow-deploy/create-service-with-flow
```
Comment (wangchao1230) [inline examples/tutorials/flow-deploy/create-service-with-flow/resources.txt:1]: too many txt files with this change. a bit ugly
image

Comment (zhengfeiwang) [reply examples/tutorials/flow-deploy/create-service-with-flow/resources.txt:1]: yes... if we can accept inconsistency between notebook and readme, how about adding a folder like `examples/tutorials/resources/`, and we can host these txts there like `examples/tutorials/resources/run-management.txt`

Comment (zhengfeiwang) [reply examples/tutorials/flow-deploy/create-service-with-flow/resources.txt:1]: for readme, it looks like below for now:

image

it's easier for me to accept this

Comment (zhengfeiwang) [reply examples/tutorials/flow-deploy/create-service-with-flow/resources.txt:1]: another proposal is to explicitly add resources in notebook/readme with special mark (e.g. `# resources`), but it's not very user friendly...

Comment (zhengfeiwang) [reply examples/tutorials/flow-deploy/create-service-with-flow/resources.txt:1]: offline synced, will integrate this into notebook/markdown and not visible when rendering.

Comment (zhengfeiwang) [reply examples/tutorials/flow-deploy/create-service-with-flow/resources.txt]: updated using metadata, may take a look

Comment (zhengfeiwang) [reply examples/tutorials/flow-deploy/create-service-with-flow/resources.txt]: btw, notebook will not render metadata, but GitHub will...

image

**Review by crazygao — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**